facades/PC/src/main/java/org/terasology/engine/GLFWSplashScreen.java

```diff
@@ -55,6 +54,47 @@ public void close() {
 
     @Override
     public void run() {
+        Renderer renderer = Mono.just(0)
+                .subscribeOn(GameScheduler.graphics())
+                .doOnNext(n -> initWindow())
+                .map((n) -> new Renderer())
+                .doOnNext(Renderer::init)
+                .doOnNext(r -> GL11.glClearColor(0f, 0f, 0f, 0f))
+                .block();
+
+        final long contextId = GameScheduler.runBlockingGraphics("steal context", GLFW::glfwGetCurrentContext);
+
+
+        Flux.interval(Duration.ofMillis(100))
+                .onBackpressureDrop()
+                .map(n -> GLFW.glfwGetTime())
+                .buffer(2)
+                .map(times -> times.get(1) - times.get(0))
+                .publishOn(GameScheduler.graphics())
+                .takeUntil(dTime -> isClosing || window.isClosing())
+                .doOnNext(dTime -> {
+                    GLFW.glfwMakeContextCurrent(contextId);
+                    renderer.clear();
+                    widgets.forEach(widget -> widget.update(dTime));
```

Comment:
oh my

does this really give you `time[n] - time[n-1]`? or is it going to give you a series more like `time[1] - time[0], time[3] - time[2], time[5] - time[4]`?

Reply:
Oh. Seems there needs to use overlaping windows..